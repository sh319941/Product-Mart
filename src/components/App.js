import React,{ Suspense, lazy } from 'react';
import { Switch, Route } from 'react-router-dom';

import AddProduct from './AddProduct';
import EditProduct from './EditProduct';
import Landing from './Landing';
import NoPageFound from './NoPageFound';
import ProductDetail from './ProductDetail';
import ProductStats from './ProductStats';
import MyProducts from "./MyProducts";
import ProfilePage from './ProfilePage';
import Cart from './Cart';
import Spinner from './ui-components/Spinner';
import { Container } from 'semantic-ui-react';

const AllProducts = lazy(()=>import('./AllProducts'));
const LoginPage = lazy(()=>import('./LoginPage'));
const RegisterPage = lazy(()=>import('./RegisterPage'));

const App = () => {

    return (
      <Container>
        <Suspense fallback={<div><Spinner /></div>}>
          <Switch>
            <Route exact path="/" component={Landing} />
            <Route exact path="/products" component={AllProducts} />
            <Route path="/login" component={LoginPage} />
            <Route path="/register" component={RegisterPage} />
            <Route path="/addProduct" component={AddProduct} />
            <Route path="/myproducts" component={MyProducts} />
            <Route path="/cart" component={Cart} />
            <Route
              path="/myprofile"
              render={(props) => {
                return (
                  <ProfilePage userId={parseInt(localStorage.getItem("id"))} />
                );
              }}
            />
            <Route
              exact
              path="/products/edit"
              render={(props) => {
                return <EditProduct product={props.location.productProps} />;
              }}
            />
            <Route
              path="/products/:name"
              render={(props) => {
                return <ProductDetail product={props.location.productProps} />;
              }}
            />
            <Route path="/productStats" component={ProductStats} />
            <Route component={NoPageFound} />
          </Switch>
        </Suspense>
      </Container>
    );
}

export default App;