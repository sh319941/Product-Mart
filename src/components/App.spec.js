import React, { Suspense } from "react";
import { shallow } from "enzyme";
import { Route } from "react-router-dom";

import App from "./App";

describe("Testing App Component", () => {
  it("Snapshot of App component", () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it("check for Routes in App", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find(Route)).toHaveLength(12);
  });
});