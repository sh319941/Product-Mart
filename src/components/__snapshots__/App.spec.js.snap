// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component Snapshot of App component 1`] = `
<Container>
  <Suspense
    fallback={
      <div>
        <Spinner />
      </div>
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/products"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        path="/login"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        path="/register"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/addProduct"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/myproducts"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/cart"
      />
      <Route
        path="/myprofile"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/products/edit"
        render={[Function]}
      />
      <Route
        path="/products/:name"
        render={[Function]}
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/productStats"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </Suspense>
</Container>
`;
